hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestCOWDataSourceStorage.scala

```diff
@@ -167,12 +167,15 @@ class TestCOWDataSourceStorage extends SparkClientFunctionalTestHarness {
     // Read Incremental Query
     // we have 2 commits, try pulling the first commit (which is not the latest)
     val firstCommit = HoodieDataSourceHelpers.listCommitsSince(fs, basePath, "000").get(0)
+    // Setting HoodieROTablePathFilter here to test whether pathFilter can filter out correctly for IncrementalRelation
+    spark.sparkContext.hadoopConfiguration.set("mapreduce.input.pathFilter.class", "org.apache.hudi.hadoop.HoodieROTablePathFilter")
```

Comment:
Here fix the test added: https://github.com/apache/hudi/pull/458

Reply:
Why are we setting a filter from the test? Isn't it supposed to be set by the Relation?

Reply:
I think the intend of the test is to test if `HoodieROTablePathFilter` is set, the incremental relation still can read the old data correctly. But this test doesn't work as our expect, as `HoodieROTablePathFilter` is not set by default.

can see if we run this test without setting pathFilter explicitly.
<img width="1508" alt="Screen Shot 2022-11-12 at 14 59 41" src="https://user-images.githubusercontent.com/10115332/201461922-b546e09d-eeaa-44c0-acdb-01f89ac830ec.png">

Reply:
I see now. Thanks for clarifying!
Shouldn't we write the test that would set `HoodieROTablePathFilter` (by using globbing for ex)

Reply:
Yea, there is a test in `org.apache.hudi.functional.TestCOWDataSource#testReadPathsOnCopyOnWriteTable` https://github.com/apache/hudi/blob/7e7b3a866b549021c5a7ad9f89f0da90aff7da68/hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestCOWDataSource.scala#L341 actually test this. `toHadoopRelation` will add the `HoodieROTablePathFilter`, and this test also contains old version files.